core/services/gateway/handler_factory.go

```diff
@@ -38,14 +42,16 @@ func NewHandlerFactory(legacyChains legacyevm.LegacyChainContainer, ds sqlutil.D
 	}
 }
 
-func (hf *handlerFactory) NewHandler(handlerType HandlerType, handlerConfig json.RawMessage, donConfig *config.DONConfig, don handlers.DON) (handlers.Handler, error) {
+func (hf *handlerFactory) NewHandler(handlerType HandlerType, handlerConfig json.RawMessage, donConfig *config.DONConfig, don handlers.DON) (job.ServiceCtx, error) {
```

Comment:
Rather than using a lowest common denominator interface to capture both, why not have both handlers we want to support implement a third interface that you can use to effectively capture the union type?

Eg.

```
type Handler interface {
   // actual implementation
   handler()
}

type Service interface {
   // actual interface
   handler()
}
```

Then you can assert that the map of handlers only contains types that implements handler(). It makes the intent a lot clearer, even if it is a bit odd at first glance.

Also I would add a comment somewhere being explicit about the supported types too :)